limiter/lifo_blocking.go

```diff
@@ -37,98 +64,49 @@ func (q *lifoQueue) len() uint64 {
 	return q.size
 }
 
-func (q *lifoQueue) push(ctx context.Context) (uint64, chan core.Listener) {
+func (q *lifoQueue) push(ctx context.Context) (func(), chan core.Listener) {
 	q.mu.Lock()
 	defer q.mu.Unlock()
-	releaseChan := make(chan core.Listener, 1)
+	releaseChan := make(chan core.Listener)
```

Comment:
remove the channel buffer since it seems like it would cause `setListener` to always return true even if the element timed out.